firebase-firestore/src/test/java/com/google/firebase/firestore/spec/SpecTestCase.java

```diff
@@ -1097,6 +1109,16 @@ public void testSpecTests() throws Exception {
       parsedSpecFiles.add(new Pair<>(f.getName(), fileJSON));
     }
 
+    String testNameFilterFromSystemProperty =
+        emptyToNull(System.getProperty(TEST_FILTER_PROPERTY));
+    Pattern testNameFilter;
```

Comment:
You should turn this into a Matcher right away, like such:

``` 
 if (testNameFilterFromSystemProperty == null) {
      testNameFilterFromSystemProperty = ".*";
 } else {
      exclusiveMode = true;
 }
 Matcher testNameMatcher = Pattern.compile(testNameFilterFromSystemProperty).matcher();
```

Reply:
A Matcher object must be created with a string against which to match.  So we must create a new Matcher object for each test name.
https://developer.android.com/reference/java/util/regex/Pattern#matcher(java.lang.CharSequence)